staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -508,6 +520,9 @@ func (r *crdHandler) updateCustomResourceDefinition(oldObj, newObj interface{})
 
 	klog.V(4).Infof("Updating customresourcedefinition %s", newCRD.Name)
 	r.removeStorage_locked(newCRD.UID)
+	if err := r.updateStorageVersionFor(newCRD); err != nil {
```

Comment:
Am a little worried to have this inside of the lock as it calls out via a client (and potentially blocks). Same above.

Reply:
discussed offline. Decoupled the update logic and the lock